GOVERNANCE.md

```diff
@@ -0,0 +1,477 @@
+  same versions, but also that dependencies should be harmonized in a way that a
+  specific problem should not be solved with more than one external library at a
+  time.
+  Updates of dependencies is automated via a github bot
+  ([dependabot](https://github.com/dependabot)).
+  To ensure harmonization of dependencies, a dedicated team (see "Team
+  Structure") is responsible for keeping an eye on the list of dependencies.
+- **License linting**
+  License linting describes the act of checking the licenses of dependencies and
+  whether they meet a certain criteria.
+  For example, it is not feasible to import an external library that is licensed
+  as GPL-3.0 in an Apache-2.0 licensed codebase.
+  Because of this, a github action is installed to lint the licenses of
+  dependencies. This action runs as a normal lint (see "evergreen master") and
+  blocks pull requests if dependencies get imported that do not meet a set of
+  rules agreed upon by the project.
+
+## Release workflow
+
+The following describes how thin-edge.io is released.
+
+The following chapters do not talk about major releases (increasing "x" in a
+"x.y.z" version number) as there is no workflow for these implemented yet.
+
+
+### Preface
+
+The following chapter describes the release process of the thin-edge.io project
+and the corrosponding crates.
+
+There are many ways software projects can be published. In small projects,
+releasing off of the `main` branch might suffice. Larger projects tend to
+implement a dedicated strategy for their release workflow.
+
+As thin-edge.io wants to become a project of potentially many individuals and
+contributing parties, we implement a dedicated release workflow as well.
+
+The goal of the workflow described in this chapter is to have _as little impact
+as possible_ on the general development of the project.
+Releases should _never_ stop, prevent or even just slow down the normal
+development of the project.
+For example, halting development on the `main` branch and forbidding
+contributors and maintainers to merge changes to it even for as little as one
+hour just so that a new release can be published is not acceptable.
+Also should "patch" releases (sometimes called "bugfix releases") have no impact
+on the project pace or even concern developers - they should just happen in
+parallel without any influence on the normal development activities.
+
+With these requirements, we implement the following workflow.
+
+### Semver conformity and API coverage
+
+We adhere to [semver](https://semver.org).
+
+It has to be noted, though, that we also follow the
+[exception in semver compatibility as defined by cargo](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio),
+that specifically means:
+
+> While SemVer says there is no compatibility before 1.0.0, Cargo considers
+> 0.x.y to be compatible with 0.x.z, where y ≥ z and x > 0.
```

Comment:
```suggestion
We adhere to [semver](https://semver.org), with the [exceptions implied by the cargo implementation of the standard](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio).
```

This can potentially be boiled down a bit to the suggested variant. No need to lose that many words over something that is clearly defined.

Reply:
Applied